### PR TITLE
chore: migrate to useRef take 2

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: packages/web/src/components/codeMirrorWrapper.tsx [hunk 1]
```diff
@@ -90,29 +90,38 @@ export const CodeMirrorWrapper: React.FC<SourceProps> = ({
...
-      updateEditor(cm, text, highlight, revealLine, focusOnChange);
       return cm;
     })();
-  }, [modulePromise, codemirror, codemirrorElement, text, language, highlight, revealLine, focusOnChange, lineNumbers, wrapLines, readOnly, onChange]);
+  }, [modulePromise, codemirror, codemirrorElement, language, lineNumbers, wrapLines, readOnly]);
+
+  React.useEffect(() => {
+    if (!codemirror)
+      return;
+    codemirror.off('change', (codemirror as any).listenerSymbol);
```
Comment (dgozman) [inline packages/web/src/components/codeMirrorWrapper.tsx:102]: Does this work when listener is `undefined`?

File: packages/web/src/components/codeMirrorWrapper.tsx [hunk 2]
```diff
@@ -90,29 +90,38 @@ export const CodeMirrorWrapper: React.FC<SourceProps> = ({
...
       });
+      codemirrorRef.current = cm;
       setCodemirror(cm);
-      if (onChange)
-        cm.on('change', () => onChange(cm.getValue()));
-      updateEditor(cm, text, highlight, revealLine, focusOnChange);
       return cm;
     })();
-  }, [modulePromise, codemirror, codemirrorElement, text, language, highlight, revealLine, focusOnChange, lineNumbers, wrapLines, readOnly, onChange]);
+  }, [modulePromise, codemirror, codemirrorElement, language, lineNumbers, wrapLines, readOnly]);
```
Comment (dgozman) [inline packages/web/src/components/codeMirrorWrapper.tsx:97]:
```suggestion
  }, [modulePromise, codemirror, codemirrorElement, language, lineNumbers, wrapLines, readOnly, codemirrrorRef]);
```

**Review by dgozman — APPROVED** [review]

File: packages/web/src/components/codeMirrorWrapper.tsx (hunk 1 above)
Comment (pavelfeldman) [reply packages/web/src/components/codeMirrorWrapper.tsx:102]: i hope so!

File: packages/web/src/components/codeMirrorWrapper.tsx (hunk 2 above)
Comment (pavelfeldman) [reply packages/web/src/components/codeMirrorWrapper.tsx:97]: codemirrorRef is const, using it in effect does not make sense.

**Pull request merged**